internal/module/module.go

```diff
@@ -39,10 +39,18 @@ func loadModule(path string) (*tfconfig.Module, error) {
 }
 
 func loadModuleItems(tfmodule *tfconfig.Module, options *Options) (*tfconf.Module, error) {
-	header, err := loadHeader(options.Path, options.HeaderFromFile)
-	if err != nil {
-		return nil, err
+	var (
+		header string
+		err    error
+	)
+
+	if options.HeaderFromFile != "" {
+		header, err = loadHeader(options.Path, options.HeaderFromFile)
```

Comment:
Considering a new properties in `Options`, I'd suggest to change to `loadHeader(options *Options)` and check the condition immediately inside the function instead of at the higher level here. The main reason is in that case unit testing the function is completely self-contained and will be accurate.